### ISSUE TITLE
Unit test components

### ISSUE THREAD

**Issue by luizbaldi** [verypossible/ReactNativeLegos#22]
In order to have a stable codebase, we need to properly unit test our core UI components.

For doing so, we're using the [`react-native-testing-library`](https://github.com/callstack/react-native-testing-library), which encourages best practices when testing React Native components. 

## List of components that need to be tested:

- [ ] `Avatar`
- [x] [`Button`](https://github.com/verypossible/ReactNativeLegos/blob/master/src/ui/components/Button/Button.spec.tsx)
- [x] [`BottomSheet`](https://github.com/verypossible/ReactNativeLegos/blob/master/src/ui/components/BottomSheet/BottomSheet.spec.tsx)
- [ ] `Control`
- [ ] `Header`
- [ ] `ListItem`
- [ ] `Modal`
- [ ] `Notification`
- [ ] `TextInput`

## Definition of done

- We don't wanna test implementation details here, so when writing tests we should focus on how the component behaves, so when props or method names change, the test must still pass
- Don't worry so much about testing coverage, it's just a metric and the important thing is to cover how the component is being actually used
- Avoid testing using snapshots, but if you **really** wanna use them, please read [this article](https://kentcdodds.com/blog/effective-snapshot-testing) before

--- 

Each component should have a separate PR, If you decide to start writing tests for some of the components listed above, please drop a comment on this issue so we can keep track of everything :)

**Comment by chrismolina159** [comment]
Hello, I would like to give this a try! Would I be able to try with the Avatar component? I'm new to unit testing, but I am interested in giving it a shot

**Comment by luizbaldi** [comment]
@chrismolina159 Of course! The `Avatar` component is a good one to start, you can basically check if style properties are being properly used.

Feel free to also refactor the component if you think there's something to improve, thanks :)

**Comment by chrismolina159** [comment]
Hello @luizbaldi  unfortunately I cannot work on this right now, so feel free to unassign me from the Avatar component.  Best of luck!

**Comment by luizbaldi** [comment]
@chrismolina159 All good, thanks for notifying! If you ever want to contribute we'll be here :)
